src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,102 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] list = (T[]) new Object[DEFAULT_CAPACITY];
+    private int capacity = DEFAULT_CAPACITY;
+    private int size = 0;
+
+    private void grow() {
+        int oldCapacity = capacity;
+        capacity = (int) (capacity * 1.5);
+        T[] growList = (T[]) new Object[capacity];
+        System.arraycopy(list, 0, growList, 0, oldCapacity);
+        list = growList;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == capacity) {
+            grow();
+        }
+        list[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index to add value!");
+        }
+        if (capacity == size) {
+            grow();
+        }
+        System.arraycopy(list, index, list, index + 1, size - index);
+        list[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        while (size + list.size() > capacity) {
+            grow();
+        }
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index > size - 1 || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index to get value!");
+        }
+        return list[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index > size - 1 || index < 0) {
```

Comment:
let's move this code to prive method checkIndex, cause it's duplicated in several methods